code/tyler/django/tyler_django_lab-2/blog_proj/users_app/views.py

```diff
@@ -0,0 +1,60 @@
+from django.shortcuts import render, redirect
+from django.urls import reverse
+from django.contrib.auth import (
+    authenticate, login as django_login, logout as django_logout)
+from django.contrib import messages
+
+from blog_app.models import BlogPost
+from .models import CustomUser
+from django.contrib.auth.decorators import login_required
+
+# Create your views here.
+
+
+def register(request):
+    if request.method == 'GET':
+        return render(request, 'users/register.html')
+    elif request.method == 'POST':
+        form = request.POST
+        username = form.get('username')
+        password = form.get('password')
+        new_user = CustomUser.objects.create_user(
+            username=username,
+            password=password
+        )
+
```

Comment:
Register should also log in the user.

Reply:
This is still not resolved. The user should get logged in after being registered and then Django should redirect to the profile view instead of rendering the profile template.